server/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Controller,
+  Post,
+  UseGuards,
+  Body,
+  HttpCode,
+  Req,
+  Get,
+} from '@nestjs/common';
+import { RegisterDto } from './auth.dto';
+import { AuthService } from './auth.service';
+import { CookieAuthGuard } from './cookieAuth.guard';
+import { LogInWithCredentialsGuard } from './logInWithCredentials.guard';
+
+@Controller('auth')
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @HttpCode(201)
+  @Post('/register')
+  async register(@Body() payload: RegisterDto) {
+    return this.authService.registerUser(payload);
+  }
+
+  @HttpCode(200)
+  @UseGuards(LogInWithCredentialsGuard)
+  @Post('/login')
+  async logIn(@Req() request) {
+    return request.user;
+  }
```

Comment:
Quick question, so once they log in the `LogInWithCredentialsGuard` will validate the request and make sure they're legit, but why is it returning user? I'm assuming the request has to match the DTO and it has email, username, and password, so should it be `request.username` or is the middleware adding this user object during login?

Reply:
Gotcha, returning the user is good, auth just makes my brain hurt trying to keep track of what goes where.